apps/emqx_plugin_libs/src/emqx_placeholder.erl

```diff
@@ -298,9 +297,12 @@ replace_with(Tmpl, RE, '$n') ->
             Parts
         ),
     Res.
-
+parse_nested(<<".", R/binary>>) ->
+    %% ignroe the root .
+    parse_nested(R);
 parse_nested(Attr) ->
     case string:split(Attr, <<".">>, all) of
+        [<<>>] -> {var, ?PH_VAR_THIS};
```

Comment:
For placeholders that contain only `<<".">>` it is `{var,?PH_VAR_THIS}`, but for empty string placeholders (`<<>>`), it should be `{var, <<>>}`.

i.e. if empty string is provided, we should send the empty string as it is to the data-bridges.

Reply:
we should probably not allow it ?

Reply:
Actually, "${}" is not a valid placeholder today. as it's parsed into :
```
> emqx_placeholder:preproc_tmpl("${}").
[{str,<<"${}">>}]
```
if we want to do something about it, it should be a separate PR.